### DISCUSSION TITLE
Testing MaterialX in Maya 2023

### DISCUSSION THREAD

**Discussion by daviddeacon** [Autodesk/maya-usd#2244] [Q&A]
I've just been testing materialX in 2023, any USD files that I import into the stage that have mtlx shaders they are completely invisible in the Maya viewport. Wireframe is visible when they are selected.

The mtlx shaders seem to be valid as they work fine in Houdini's solaris. 

I'm on macOS 12.2.

Is this expected behaviour currently?

Thanks,
David

**Comment by santosd** [comment]
Hi @daviddeacon wondering if you have any files you can share with us, so that we can take a look. Thank you.
-David Santos

**Reply by daviddeacon** [reply]
Sure,
Exported from Maya -
https://www.dropbox.com/s/b51b3qyp4py3amj/torus_maya2023.usda?dl=0
Exported from Solaris -
https://www.dropbox.com/s/dnfp0p9dwrxhxj5/torus_solaris.usda?dl=0

They both look like this as part of the USD stage.
https://www.dropbox.com/s/rdfywpwcv7xiehl/Screenshot%202022-04-05%20at%2020.43.22.png?dl=0

They both load back into solaris correctly.

Thanks,
David.

**Comment by daviddeacon** [comment]
I just grabbed the installer for 0.17.0 for 2022. Unfortunately the same problem there :(

**Comment by santosd** [comment]
Hi David,
Thank you for the files. I tested this on my end and did find that there seems to be an issue in Maya 2022. I am testing on Windows and don't see the issue where the prims are invisible. However, they are not displaying the material like they should. It seems to be a problem with the MaterialX libraries and how they install on the Maya USD plugin for 2022. I also tested in Maya 2023 using 0.17.0 which is now available on [GitHub](https://github.com/Autodesk/maya-usd/releases/tag/v0.17.0) and I was able to get it working there. Please give that a try and see if that is working for you.

I will let you know what I find in regards to 2022.

Cheers,
David Santos

**Comment by daviddeacon** [comment]
I've also tested 0.17.0 in Maya 2023, same invisible prims on macOS.

I guess this should now be raised an issue?

**Comment by santosd** [comment]
Yes, I will convert it over to an issue.

**Comment by santosd** [comment]
Created this issue for it and have logged this internally: https://github.com/Autodesk/maya-usd/issues/2267
